pkg/virt-launcher/virtwrap/manager.go

```diff
@@ -320,34 +324,61 @@ func (l *LibvirtDomainManager) FinalizeVirtualMachineMigration(vmi *v1.VirtualMa
 	return l.finalizeMigrationTarget(vmi)
 }
 
-// HotplugHostDevices attach host-devices to running domain
-// Currently only SRIOV host-devices are supported
+// HotplugHostDevices attach host-devices to running domain, currently only SRIOV host-devices are supported.
+// This is operation runs on the background each at a time.
 func (l *LibvirtDomainManager) HotplugHostDevices(vmi *v1.VirtualMachineInstance) error {
+	if l.hotplugHostDevicesInProgress.Load().(bool) {
+		log.Log.V(3).Object(vmi).Info("hot-plug host-devices is in progress")
+		return nil
+	}
+
+	l.asyncHotplugHostDevices(vmi)
+
+	return nil
+}
+
+func (l *LibvirtDomainManager) asyncHotplugHostDevices(vmi *v1.VirtualMachineInstance) {
+	l.hotplugHostDevicesInProgress.Store(true)
+
+	go func() {
+		defer l.hotplugHostDevicesInProgress.Store(false)
+
+		log.Log.V(3).Object(vmi).Info("starting hot-plug host-devices")
+
+		if err := l.hotplugHostDevices(vmi); err != nil {
+			log.Log.Object(vmi).Error(err.Error())
+		}
+	}()
+}
+
+func (l *LibvirtDomainManager) hotplugHostDevices(vmi *v1.VirtualMachineInstance) error {
```

Comment:
Can you please rename this method so it won't be so similar to "HotplugHostDevices"?